### PR TITLE
apply normalisation for every vnode

### PR DESCRIPTION
**Pull request by JoviDeCroock**
fixes: https://github.com/developit/preact/issues/1445

### CODE REVIEW

**Comment by JoviDeCroock** [comment]
Coverall seems to bug on my PR's sometimes, also still getting spammed quite hard by travis.

**Comment by JoviDeCroock** [comment]
It could be a feature to make it so that we have options.stateBailoutAlgo or something along those lines.

**Comment by JoviDeCroock** [comment]
we have a test for that first case where we check if onDoubleClick and onBeforeInput are normalized. Am I misunderstanding this or?

File: test/browser/compat.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { createElement, render } from 'preact';
+import { setupScratch, teardown } from '../../test/_util/helpers';
+
+import React from 'preact/compat'; // eslint-disable-line
+
+describe('imported compat', () => {
```
Comment (marvinhagemeister) [inline test/browser/compat.test.js]: Since these tests are for compat, we should move them to the compat test suite :+1:

**Comment by marvinhagemeister** [comment]
That's strange: Looking at the output in our CI it seems like it doesn't run all tests and aborts somewhere inside Lifecylce tests. Past builds seem to halt at the same position, but no error code is thrown or anything. The test suite returns a proper `exit 0` code. Because a lot of tests are not run the coverage will fail. Not sure what's causing this.

File: test/browser/compat.test.js (hunk 1 above)
Comment (JoviDeCroock) [reply test/browser/compat.test.js]: moved file

**Comment by JoviDeCroock** [comment]
Yes saw so aswell :/ Don't understand why though

**Pull request merged**